testgen/src/validator_set.rs

```diff
@@ -0,0 +1,88 @@
+use crate::{
+    helpers::*,
+    validator::{generate_validators, Validator},
+    Generator,
+};
+use gumdrop::Options;
+use serde::{Deserialize, Serialize};
+use simple_error::*;
+use tendermint::validator;
+
+#[derive(Debug, Options, Serialize, Deserialize, Clone)]
+pub struct ValidatorSet {
+    #[options(
+        parse(try_from_str = "parse_as::<Vec<Validator>>"),
+        help = "validators (required)"
+    )]
+    validators: Option<Vec<Validator>>,
```

Comment:
could a validator set ever be empty?

Reply:
Ideally, not. But in testgen we want to allow such situations to be able to test them.

Reply:
yes, as this is primarily for testing, we do want to test situations when the validator set coming from the wire is empty.